ui/imports/shared/popups/CommunityMembershipSetupDialog.qml

```diff
@@ -87,23 +88,20 @@ StatusStackModal {
 
     finishButton: StatusButton {
         interactive: {
-            if (root.isInvitationPending)
+            if (root.isInvitationPending || d.accessType !== Constants.communityChatOnRequestAccess)
                 return true
 
-            if (root.requirementsCheckPending || !root.joinPermissionsCheckSuccessful)
+            if (root.checkingPermissionToJoinInProgress || !root.joinPermissionsCheckSuccessful)
```

Comment:
`checkingPermissionToJoinInProgress` doesn't seem necessary. Can't you just use `joinPermissionsCheckSuccessful`?

`joinPermissionsCheckSuccessful` is also only for the Join permissions, since we have `channelsPermissionsCheckSuccessful` for the channels.

So instead of checking `checkingPermissionToJoinInProgress`, just check `!joinPermissionsCheckSuccessful`

Reply:
Ah I guess for the loading it might be better to use `checkingPermissionToJoinInProgress`. So we can keep it

Reply:
Yeah, I need it for the loading and then basically enable the button when:
`!checkingPermissionToJoinInProgress && joinPermissionsCheckSuccessful` (I need both of these)